File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/661_:::_Blowing_Fuses/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		int caso=1, n, m, c, mpc, max, lect;
		for(String linea, aux[]; !(linea=br.readLine()).equals("0 0 0"); caso++)
		{
			aux=linea.split(" ");
			n=Integer.parseInt(aux[0]);
			m=Integer.parseInt(aux[1]);
			c=Integer.parseInt(aux[2]);

			int[][] amp=new int[n][2];
			mpc=max=0;

			for(int i=0; i<n; i++)
			{
				amp[i][0] = Integer.parseInt(br.readLine());
				amp[i][1] = 0;
			}

			while(m--!=0)
			{
				lect=Integer.parseInt(br.readLine())-1;
				if(amp[lect][1] == 0)
				{
					mpc += amp[lect][0];
					amp[lect][1] = 1;
				}
				else
				{
					mpc -= amp[lect][0];
					amp[lect][1] = 0;
				}
				if(max < mpc) max = mpc;
			}

			System.out.println("Sequence "+caso);
			if(max > c) System.out.println("Fuse was blown.");
			else
			{
				System.out.println("Fuse was not blown.");
				System.out.println("Maximal power consumption was "+max+" amperes.");
			}
			System.out.println();
		}
		//-----------------------------------------------------------
		br.close();
	}
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/661_:::_Blowing_Fuses/Main.cpp
using namespace std;
#include <cstdio>

int main()
{
	int caso, n, m, c, mpc, max, lect, i;
	int amp[20][2];
	for(caso=1; scanf("%d %d %d", &n, &m, &c),(n||m||c); caso++)
	{
		mpc=max=0;

		for(i=0; i<n; i++)
		{
			scanf("%d", &amp[i][0]);
			amp[i][1] = 0;
		}

		while(m--)
		{
			scanf("%d", &lect);
			lect--;
			if(amp[lect][1] == 0)
			{
				mpc += amp[lect][0];
				amp[lect][1] = 1;
			}
			else
			{
				mpc -= amp[lect][0];
				amp[lect][1] = 0;
			}
			if(max < mpc) max = mpc;
		}

		printf("Sequence %d\n", caso);
		if(max > c) puts("Fuse was blown.");
		else
		{
			puts("Fuse was not blown.");
			printf("Maximal power consumption was %d amperes.\n", max);
		}
		puts("");
	}
	
	return 0;
}
